### ISSUE TITLE
🚀 Feature: Database Provisioning

### ISSUE THREAD

**Issue by maximilianmaihoefner** [appwrite/appwrite#2429]
### 🔖 Feature description

A way to provision Database (Rules) via YAML, JSON or similar, to enable Infrastructure as Code possibilities.

### 🎤 Pitch

Let's say you have two apps in Appwrite, one for your Production- and one for QA/Testing-Environment. If the Database Rules could be defined in some kind of Text-Resource and applied to the Appwrite-App via CLI/Rest/etc. you could test your rules configuration on QA and, when ready, apply the same rules to Production, cutting out the possibility for human-error when creating, updating or deleting rules.

Note: This might already be somewhat possible using the appwrite-cli, but I am unsure about the syntax. The docs only have this example:
```
appwrite database updateCollection \
        --collectionId="[COLLECTION_ID]" \
        --name="[NAME]" \
        --read="" \
        --write="" \
        --rules=""
```
But when I used this command: `appwrite database updateCollection --collectionId=<idOfCollection> --name=<nameOfCollection> --rules="{ \"label\": \"Name\", \"key\": \"name\", \"type\": \"text\", \"default\": \"\", \"required\": \"false\", \"array\": \"false\" }"
`

I received:
```
message : Server Error
code : 500
version : 0.11.0
```
and the logs state:
```
appwrite                        | [Error] Timestamp: 2021-11-21T10:29:39+00:00
appwrite                        | [Error] Method: PUT
appwrite                        | [Error] URL: /v1/database/collections/:collectionId
appwrite                        | [Error] Type: TypeError
appwrite                        | [Error] Message: array_merge(): Argument #2 must be of type array, string given
appwrite                        | [Error] File: /usr/src/code/src/Appwrite/Database/Validator/Collection.php
appwrite                        | [Error] Line: 45
```
Trying with `--rules=[...]` results in a zsh error `zsh: no matches found: --rules=[...]`

Regardless of the apprwrite-cli issue, it would be nice if there was a way to pass a file for rules, or even a definition for the entire collection (id, name, rules, permissions, etc.)

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by eldadfux** [comment]
Allowing setting collections, attributes, and indexes will be available with the future implementation of the `appwrite.json` file. This implementation is planned under the RFC of the CLI 2.0 which is already in progress.

You can learn more here:
https://github.com/appwrite/rfc/blob/cli-2.0/018-cli-2.0/018-cli-2.0.md#-the-appwritejson-file

**Comment by TorstenDittmann** [comment]
We just released 0.13 with a new CLI and this possible now 👍🏻 

You can find the announcement [here](https://dev.to/appwrite/announcing-appwrite-013-with-major-upgrades-to-storage-and-functions-3hpf).

Thank you a lot for reporting this 🙏🏻
